### PR TITLE
Andrsmith/add automanage snippets

### PR DESCRIPTION
**Pull request by AndrewCS149**
Added: 
- Automanage Best Practices DevTest Assignment.snippet.json
- Automanage Best Practices Prod Assignment.snippet.json
- Automanage Custom Profile Assignment.snippet.json
- Automanage Create Custom Profile.snippet.json

### CODE REVIEW

**Comment by AndrewCS149** [comment]
Sounds great @StephenWeatherford!

**Comment by StephenWeatherford** [comment]
Running suites locally because they unfortunately don't all run for remote branches...

**Comment by StephenWeatherford** [comment]
I need to update schemas...
image

**Comment by StephenWeatherford** [comment]
After updating schemas, still missing 3 of the 4:
image

File: assets/resourceSnippets/Automanage Best Practices DevTest Assignment.snippet.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (StephenWeatherford) [inline assets/resourceSnippets/Automanage Best Practices DevTest Assignment.snippet.json:1]: Please make the following change to test/functional/snippets.test.ts in order to turn off schema validation during tests for these snippets:

```ts
// Override whether to ignore schema validation - default is to perform schema validation - mark with "true" to ignore schema validation
// TODO: All items in this list indicate an error (either snippet or schema) and should eventually be removed
const overrideIgnoreSchemaValidation: { [name: string]: boolean } = {
    /* // TODO: fix snippet
     +   'Missing required property "protectedSettings"'
     */
    "Linux VM Custom Script": true,

==>>> ADD THESE LINES:
    // TODO: Remove when schemas are published
    "Automanage Best Practices DevTest Assignment": true,
    "Automanage Best Practices Prod Assignment": true,
    "Automanage Create Custom Profile": true,
    "Automanage Custom Profile Assignment": true,
};
```

File: assets/resourceSnippets/Automanage Best Practices DevTest Assignment.snippet.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "metadata": {
+        "prefix": "arm-automanage-best-practices-devtest-assignment",
```
Comment (StephenWeatherford) [inline assets/resourceSnippets/Automanage Best Practices DevTest Assignment.snippet.json:5]: The prefix is generally kept relatively short.  Users can still see and search on the description.
```suggestion
        "prefix": "arm-automanage-bp-devtest-assignment",
```
image

File: assets/resourceSnippets/Automanage Best Practices Prod Assignment.snippet.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "metadata": {
+        "prefix": "arm-automanage-best-practices-prod-assignment",
```
Comment (StephenWeatherford) [inline assets/resourceSnippets/Automanage Best Practices Prod Assignment.snippet.json:5]:
```suggestion
        "prefix": "arm-automanage-bp-prod-assignment",
```

File: assets/resourceSnippets/Automanage Best Practices DevTest Assignment.snippet.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+        "description": "Create a Best Practices DevTest Configuration Profile Assignment"
+    },
+    "resources": [
+        {
+            "type": "Microsoft.Compute/virtualMachines/providers/configurationProfileAssignments",
```
Comment (StephenWeatherford) [inline assets/resourceSnippets/Automanage Best Practices DevTest Assignment.snippet.json:10]: @bmoore-msft Is it better to use a scope property like the following that I compiled from the bicep in https://docs.microsoft.com/en-us/azure/azure-resource-manager/bicep/scope-extension-resources#apply-to-resource?
```json
    {
      "type": "Microsoft.Authorization/locks",
      "apiVersion": "2016-09-01",
      "scope": "[format('Microsoft.Storage/storageAccounts/{0}', 'examplestore')]",
      "name": "storeLock",
      "properties": {
        "level": "CanNotDelete",
        "notes": "Storage account should not be deleted."
      }
    }

```

File: assets/resourceSnippets/Automanage Custom Profile Assignment.snippet.json [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+        "prefix": "arm-automanage-custom-assignment",
+        "description": "Create a Custom Automanage Configuration Profile Assignment"
+    },
+    "resources": [
+       {
+            "type": "Microsoft.Compute/virtualMachines/providers/configurationProfileAssignments",
+            "apiVersion": "2022-05-04",
+            "name": "[concat('${1:vmName}', '/Microsoft.Automanage/default')]",
+            "properties": {
+                "configurationProfile": "/subscriptions/${2:subscriptionId}/resourceGroups/${3:resourceGroup}/providers/Microsoft.Automanage/configurationProfiles/${4:profileName}"
```
Comment (StephenWeatherford) [inline assets/resourceSnippets/Automanage Custom Profile Assignment.snippet.json:14]: Please use the resourceId function instead.  See https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-test-cases#use-resource-id-functions

**Review by StephenWeatherford — CHANGES_REQUESTED** [review]
See comments thx

File: assets/resourceSnippets/Automanage Best Practices DevTest Assignment.snippet.json (hunk 1 above)
Comment (AndrewCS149) [reply assets/resourceSnippets/Automanage Best Practices DevTest Assignment.snippet.json:1]: Done

File: assets/resourceSnippets/Automanage Best Practices Prod Assignment.snippet.json (hunk 3 above)
Comment (AndrewCS149) [reply assets/resourceSnippets/Automanage Best Practices Prod Assignment.snippet.json:5]: Done

File: assets/resourceSnippets/Automanage Custom Profile Assignment.snippet.json (hunk 5 above)
Comment (AndrewCS149) [reply assets/resourceSnippets/Automanage Custom Profile Assignment.snippet.json:14]: Does this look right? 

```json
"configurationProfile": "[resourceId('${2:subscriptionId}', '${3:resourceGroup}', 'Microsoft.Compute/virtualMachines/providers/configurationProfileAssignments', '${1:vmName}/Microsoft.Automanage/default')]"
```

File: assets/resourceSnippets/Automanage Best Practices DevTest Assignment.snippet.json (hunk 1 above)
Comment (StephenWeatherford) [inline assets/resourceSnippets/Automanage Best Practices DevTest Assignment.snippet.json:1]: Could you please merge this PR?  Thx 
https://github.com/AndrewCS149/vscode-azurearmtools/pull/1

File: assets/resourceSnippets/Automanage Best Practices DevTest Assignment.snippet.json (hunk 2 above)
Comment (AndrewCS149) [reply assets/resourceSnippets/Automanage Best Practices DevTest Assignment.snippet.json]: done

File: assets/resourceSnippets/Automanage Custom Profile Assignment.snippet.json (hunk 5 above)
Comment (StephenWeatherford) [reply assets/resourceSnippets/Automanage Custom Profile Assignment.snippet.json:14]: Can you test it with a deployment?  thx.

File: assets/resourceSnippets/Automanage Best Practices DevTest Assignment.snippet.json (hunk 4 above)
Comment (bmoore-msft) [reply assets/resourceSnippets/Automanage Best Practices DevTest Assignment.snippet.json:10]: using `scope` is the pattern we've been migrating to...

File: assets/resourceSnippets/Automanage Custom Profile Assignment.snippet.json (hunk 5 above)
Comment (AndrewCS149) [reply assets/resourceSnippets/Automanage Custom Profile Assignment.snippet.json]: alright ive made the update and verified that it works.

File: assets/resourceSnippets/Automanage Best Practices DevTest Assignment.snippet.json (hunk 4 above)
Comment (AndrewCS149) [reply assets/resourceSnippets/Automanage Best Practices DevTest Assignment.snippet.json]: @StephenWeatherford Ive applied the scope property and verified that it deploys correctly. Does it look correct to you? 

- Automanage Best Practices DevTest Assignment.snippet.json
- Automanage Best Practices Prod Assignment.snippet.json
- Automanage Create Custom Profile.snippet.json

File: assets/resourceSnippets/Automanage Best Practices DevTest Assignment.snippet.json [hunk 6]
```diff
@@ -7,9 +7,10 @@
     },
     "resources": [
         {
-            "type": "Microsoft.Compute/virtualMachines/providers/configurationProfileAssignments",
+            "type": "Microsoft.Automanage/configurationProfileAssignments",
+            "scope": "'Microsoft.Compute/virtualMachines/${1:vmName}",
```
Comment (bmoore-msft) [inline assets/resourceSnippets/Automanage Best Practices DevTest Assignment.snippet.json:11]: Can you move the `scope` property above the `type` property - a little more readable and the common patther.

Comment (AndrewCS149) [reply assets/resourceSnippets/Automanage Best Practices DevTest Assignment.snippet.json:10]: Done

**Review by StephenWeatherford — APPROVED** [review]
Looks great, thanks!

**Pull request merged**